### DISCUSSION TITLE
Creating/uploading new torrents

### DISCUSSION THREAD

**Discussion by ATL-Flaneur** [haugene/docker-transmission-openvpn#2056] [Q&A] [answered]
How can I add and publish a torrent using this container? That is, point it at a new file to share and get a magnet URL? I'm not seeing an obvious route through the web interface. Is this something where I need to connect to the container?

**Comment by haugene** [comment, marked as answer]
Hey Andreas :wave: And thank you for a generous donation :pray: 

It does not seem like creating torrents is possible through the Transmission Web UI. But you can definitely do this using the command line from within the container.
If you already have the container running on your machine/server you can run `docker ps` in a shell/terminal and then find the transmission-openvpn container and exec into it with `docker exec -it <name or id> bash`.

At this point you can create a torrent file from any file or folder you've mounted into it. Assuming that you've mounted your files at "/data/" which is the default and you have `some.file` there that you want to make a torrent for you can run:
```
transmission-create -o /data/my.torrent /data/some.file
```

I'm not too familiar with making torrent files but it also seems that you should have a tracker announce URL that you can specify like this:
```
transmission-create -o /data/my.torrent -t udp://tracker.openbittorrent.com:80 /data/some.file
```

What different trackers need for you to use them, I don't know. I've just stitched together examples and stuff I found online.

FYI:
You don't have to have the container running already on your machine. If you have the file you want to make a torrent of locally in `/some/folder/` you can simply run:
```
docker run --rm -it -v /some/folder/:/data haugene/transmission-openvpn bash -c "transmission-create -o /data/my.torrent /data/some.file"
```
This will fire up a one-off container that just creates the torrent within whatever folder you mount as data. Paths needs to be updated to point to whatever you want the output to be named and the torrent to contain.

Docs on the transmission-create cli tool can be found here: http://manpages.ubuntu.com/manpages/trusty/man1/transmission-create.1.html

**Reply by ATL-Flaneur** [reply]
Got it working! After `transmission-create`, you need to add:

`transmission-remote --add /data/my.torrent`

This tells the Transmission daemon to load the file and begin seeding.
